app/models/bookmark.rb

```diff
@@ -21,6 +21,19 @@ def not_already_bookmarked_by_user
     errors.add(:base, ts("You have already bookmarked that."))
   end
 
+  validate :check_new_external_work
+  def check_new_external_work
+    return unless bookmarkable.is_a?(ExternalWork) && bookmarkable.new_record?
+
+    errors.add(:base, "Fandom tag is required") if bookmarkable.fandom_string.blank?
```

Comment:
Should we make this a validation on ExternalWork?

Reply:
The `ExternalWork`s used for `RelatedWork`s don't have an associated fandom, so it'd be tricky to make it a validation directly on `ExternalWork`s. I think I'd have to add an `attr_accessor` indicating whether the `ExternalWork` was being created from a `Bookmark` or a `RelatedWork`, and then have the validation be conditional.

Reply:
Ah, thanks! `RelatedWork` is what I'm missing. The current setup is fine.